ch6/boggle.py

```diff
@@ -0,0 +1,47 @@
+board = []
+dx = [-1, -1, -1, 1, 1, 1, 0, 0]
+dy = [-1, 0, 1, -1, 0, 1, -1, 1]
+
+def inRange(y, x):
+    if x < 0 or y < 0 or x > 4 or y > 4:
+        return False
+    return True
+
+
+def hasWord(y, x, word):
+    if not inRange(y, x):
+        return False
+    if board[y][x] != word[0]:
+        return False
+    if len(word) == 1:
+        return True
+    for direction in xrange(8):
+        nextY = y + dy[direction]
+        nextX = x + dx[direction]
+        if hasWord(nextY, nextX, word[1:]):
+            return True
+    return False
+
+
+C = input()
+for _ in xrange(C):
+    for i in xrange(5):
+        a = list(raw_input())
+        board.append(a)
```

Comment:
board.append(raw_input().strip()) ? because a is never used. raw_input() probably has a trailing '\n'. e.g. "URLPM\n"